src/SIL.Harmony/SnapshotWorker.cs

```diff
@@ -242,13 +242,13 @@ private void AddSnapshot(ObjectSnapshot snapshot)
     private bool IsNew(ObjectSnapshot snapshot)
```

Comment:
so this should probably check both root and pending snapshots to see if the snapshot we want exists, just because the entity id is in pending doesn't mean the snapshot we want is not in root.

Reply:
Yes, there could be a snapshot for the entity in both root and pending, but we only ever call this method with `prevSnapshot`. So, we only need to compare the snapshot to the most recent snapshot, which is the one in pending if there is one.

If we really wanted this method to do a true `IsNew` check, it would have to look at intermediate snapshots as well. 🤷 As it stands, it's basically just the inverse of `GetSnapshot`, which prefers pending over root.

I can change it if you really want, but there shouldn't be any practical effect and I think this code is on the way out anyway.